pkg/provision/engine.go

```diff
@@ -77,17 +81,23 @@ type EngineOps struct {
 // one reservation at a time. On error, the engine will log the error. and
 // continue to next reservation.
 func New(opts EngineOps) *Engine {
+	memStats, err := mem.VirtualMemory()
+	if err != nil {
```

Comment:
If this fail, the code at line `100` will panic with nil pointer deference error. May be it's better if New returns an error also then the caller can decide to exit the program